pkg/hive/health/command.go

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package health
+
+import (
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	healthPkg "github.com/cilium/cilium/pkg/health/client"
+	"github.com/cilium/cilium/pkg/hive/health/types"
+	"github.com/cilium/hive"
+	"github.com/cilium/hive/script"
+	"github.com/cilium/statedb"
+
+	"github.com/spf13/pflag"
+)
+
+func healthCommands(db *statedb.DB, table statedb.Table[types.Status]) hive.ScriptCmdsOut {
+	return hive.NewScriptCmds(map[string]script.Cmd{
+		"health":    healthTreeCommand(db, table),
+		"health/ok": allOK(db, table),
+	})
+}
+
+func healthTreeCommand(db *statedb.DB, table statedb.Table[types.Status]) script.Cmd {
+	return script.Command(
+		script.CmdUsage{
+			Summary: "List registered metrics",
+			Args:    "[reporter-id-prefix]",
+			Flags: func(fs *pflag.FlagSet) {
+				fs.StringP("match", "m", "", "Output only health reports where the reporter ID path contains the substring")
+				fs.StringArrayP("levels", "s", []string{types.LevelOK, types.LevelDegraded, types.LevelDegraded},
+					"Output only health reports with the specified state (i.e. ok,degraded,stopped)")
+				fs.StringP("file", "f", "", "File to write output to")
+			},
+			Detail: []string{
+				"Prints out a health reporter tree.\n",
+				"If passed prefix is not-empty then only nodes of this subtree\n",
+				"will be displayed\n",
+			},
+		},
+		func(s *script.State, args ...string) (script.WaitFunc, error) {
+			var prefix string
+			if len(args) > 0 {
+				prefix = args[0]
+			}
+
+			match, err := s.Flags.GetString("match")
+			if err != nil {
+				return nil, err
+			}
+
+			levels, err := s.Flags.GetStringArray("levels")
+			if err != nil {
+				return nil, err
+			}
+
+			file, err := s.Flags.GetString("file")
+			if err != nil {
+				return nil, err
+			}
+
+			for i := range levels {
+				levels[i] = strings.ToLower(levels[i])
+			}
+
+			w := s.LogWriter()
```

Comment:
Hmm I wonder if it'd be better to return a `WaitFunc` and write to the `stdout string` return value instead. This way we can do:
```
health
stdout "DEGRADED"
```

On the other hand, I'm not sure if we want to use the `health` command in tests to assert or whether it's then preferred to use `db/cmp health` and so on. If we want `health` to be useful in tests then a one-liner output format would be useful too (though again, `db/show health` is that...)

Reply:
On the other hand, I'm not sure if we want to use the health command in tests to assert or whether it's then preferred to use db/cmp health and so on.

Yeah I would consider using this command for assertions to be bad practice.  Would mean a lot of annoying test fixing if we decide to change the tree character formatting 😄

Reply:
Actually, after starting to use the waitFunc faculty here - I've actually come to the opinion that this is a good use case for the log writer as it discourages writing test assertions (we'll make an exception for pkg/hive/health 😅).  

Log writer on the other hand is meant for displaying use facing debug output I would think which is basically the use case here (and we'll leave the output flag just so we can easily test changes to the tree rendering code).